### PR TITLE
UI/chat histories

### PR DESCRIPTION
**Pull request by osala-eng**
## Description
UI for chat history threads.

Fixes/Implements  #151 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactor and code improvement (non-breaking change which improves code quality and/or performance)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation
- [ ] Tests
- [ ] Other chores such as maintenance

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration by modifying the list below.

- [ ] Test A
- [ ] Test B
- [ ] Test C

**Test Configuration**:

Please describe the test setup. List them below as bullet points.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings
- [ ] The commit message follows the convention of this project

### CODE REVIEW

File: web/utils/handlers.py [hunk 1]
```diff
@@ -902,3 +903,29 @@ def handle_public_session() -> None:
...
+            first_query = thread_summery[0][1]
+            topic = first_query[:100]
+            run_queries.update_thread_topic(topic, feature, thread_id)
+    return topic
+
+
+def handle_get_chat_history_threads(feature: domain.FeatureKey) -> List[Tuple[int, str, datetime]]:
+    """Get chat history threads."""
+    threads = run_queries.list_thread_history(feature)
+    return [ (t[0], _create_topic_summery(t[1], feature, t[0]), t[2]) for t in threads ] # type: ignore
```
Comment (janaka) [inline web/utils/handlers.py:923]: This looks like it regenerates the thread topic everytime the list is generated?

File: source/docq/run_queries.py [hunk 2]
```diff
@@ -67,22 +67,22 @@ def _save_messages(data: list[tuple[str, bool, datetime, int]], feature: Feature
...
+            cursor.execute(f"INSERT INTO {tablename} (message, human, timestamp, thread_id) VALUES (?, ?, ?, ?)", x)  # noqa: S608
             rows.append((cursor.lastrowid, x[0], x[1], x[2], x[3]))
         connection.commit()
 
     return rows
 
 
 def _retrieve_messages(
-    cutoff: datetime, size: int, feature: FeatureKey, thread_id: int
+    cutoff: datetime, size: int, feature: FeatureKey, thread_id: int, invert: bool = False
```
Comment (janaka) [inline source/docq/run_queries.py:78]: Why do we need 'invert' as a param to control?

**Review by janaka — COMMENTED** [review]
Left a couple of comments.

**Comment by osala-eng** [comment]
#### Sample screenshot

image

**Review by janaka — APPROVED** [review]

**Pull request merged**